vslib/inc/MACsecAttr.h

```diff
@@ -0,0 +1,37 @@
+#pragma once
+
+#include "HostInterfaceInfo.h"
+
+#include <string>
+#include <memory>
+
+namespace saivs
+{
+
+    using macsec_sci_t = std::string;
+    using macsec_an_t = std::uint16_t;
+    using macsec_pn_t = std::uint64_t;
+
+    struct MACsecAttr
+    {
+        std::string m_veth_name;
+        std::string m_macsec_name;
+        std::string m_auth_key;
+        std::string m_sak;
+        std::string m_sci;
+        macsec_an_t m_an;
+        macsec_pn_t m_pn;
+        bool m_send_sci;
+        bool m_encryption_enable;
+        sai_int32_t m_direction;
+        std::shared_ptr<HostInterfaceInfo> m_info;
+
+        // Explicitely declare constructor and destructor as non-inline functions
+        // to avoid 'call is unlikely and code size would grow [-Werror=inline]'
+        // in "-O3" optimization
+        MACsecAttr();
+
+        ~MACsecAttr();
```

Comment:
I wanted to declare them as  "= default", but it will raise a compile error 'call is unlikely and code size would grow'. Because '= default' will be the inline function by default.

In O3 optimization, a variable with inline ctr/dtr will be rejected if it's a local variable in a function.

 Here is some hint I found: https://stackoverflow.com/questions/36921744/gcc-o3-flag-causes-winline-call-is-unlikely-and-code-size-would-grow-warning . I don't know what is the graceful method to fix it. So I just choose to explicitly declare ctr/dtr to fix this issue. Do you have any suggestion?